server.js

```diff
@@ -45,157 +53,164 @@ function main () {
+          }
+          resolve(championsMap);
         }
         else {
-            if (error && response) {
-                console.log("Problem with the request, error is: " + error + " with status code " + response.statusCode);
-            }
+          console.log("ERROR");
+          if (error && response) {
+            console.log("Problem with the request, error is: " + error + " with status code " + response.statusCode);
+          }
+          return Promise.resolve("Fail");
         }
+      });
     });
+  });
+
+  var all = b.then(function(val) {
+    //console.log(val);
+  });
+  return Promise.all([a, b]);
+
 }
 
-function getChampionsJson (ApiKey, toSend, championsMap, region, summonerName, version, callback) {
-    let req = "http://ddragon.leagueoflegends.com/cdn/" + version + "/data/en_US/champion.json " ;
-    request(req, function (error, response, body) {
-        if (!error && response.statusCode == 200) {
+function getVersion (region) {
+    let req = "https://global.api.pvp.net/api/lol/static-data/" + region + "/v1.2/versions?api_key=" + ApiKey;
+    return rp({uri: req, json: true})
+      .then(function (data) {
+        let version = "";
+        version = data[0];
+        return Promise.resolve(version);
+      })
+      .catch(function(e) {
+        console.log("get version request failed");
+        return Promise.resolve("Request Fail");
+      })
+    ;
+}
+
+function getPlayerInfo(region, summonerName) {
+  let a = getSummonerId(region, summonerName);
+  let b = a.then(function(infos) {
+    let req = "https://" + region + ".api.pvp.net/championmastery/location/" + PLATFORMS[region] + "/player/" + infos.id + "/champions?api_key=" + ApiKey;
+    return new Promise (function(resolve, reject) {
+        request(req, function (error, response, body) {
+          if (!error && response.statusCode == 200) {
             let str = JSON.parse(body);
-            delete str.type;
-            delete str.format;
-            delete str.version;
-            str = str.data;
-            for (let champ in str) {
-                let champTmp = [];
-                champTmp.push(str[champ].id);
-                champTmp.push(str[champ].name);
-                championsMap[str[champ].key] = champTmp;
+            for (let i = 0; i < str.length; ++i) {
+              delete str[i].playerId;
+              delete str[i].lastPlayTime;
+              delete str[i].championPointsSinceLastLevel;
             }
-            callback(null, ApiKey, toSend, championsMap, region, summonerName, version);
-        }
-        else {
-            if (error && response) {
-                console.log("Problem with the request, error is: " + error + " with status code " + response.statusCode);
+            resolve(str);
+          }
+          else {
+            if (error && response){
+              console.log("Problem with the request, error is: " + error + " with status code " + response.statusCode);
```

Comment:
Reject the promise here, or else if you have an error the promise will never resolve and everything will be stuck.